.github/workflows/ci-macos-arm64.yml

```diff
@@ -0,0 +1,95 @@
+name: CI macOS ARM64
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+      - '**.rst'
+      - '**.txt'
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+
+  build:
+    name: CI on macOS
+
+    runs-on: macos-14
+
```

Comment:
Do you think we could avoid some duplication by having a single `ci-macos.yml` with a few conditional steps?

Reply:
Ok. I've rewritten it in a single file as you advised.